migration/migration-commands/src/main/java/ddf/migration/commands/MigrateCommand.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.migration.commands;
+
+import ddf.migration.api.DataMigration;
+import ddf.migration.api.ServiceNotFoundException;
+import java.util.Collection;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.codice.ddf.commands.catalog.SubjectCommands;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+@Command(scope = "migrate-data", name = "run", description = "Runs a specific data migration task")
```

Comment:
There exists a `migrate` command in DDF. This PR will add a `migrate-data` command. I find that confusing. Is there a way we can make a clearer distinction? Or better, convert the other migration command to be a "data migration task" that can be invoked with this command.

Reply:
The `MigrateCommand` seems like more ceremony than necessary to invoke the method `migrate()`. You could do the same thing from the Karaf Console without the `MigrateCommand`:

```
admin@root> workspaceQueryRef = ($.context getServiceReferences ddf.migration.api.DataMigration "(id=workspace-query)") 0
admin@root> ($.context getService $sref) migrate
```
That way you could elminate a class.

Reply:
What the consequences are of not migrating workspace metacards? What would happen if operators did not run this command? Would anything blow up? If something would blow up, a manual migration might not be the best way to do this. Instead, a **post-query plugin** just for Workspace Metacards might be more appropriate. 

The plugin would do execute the logic in `migrateWorkspaceMetacard`to create the Query Metacards if there were unmigrated Workspaces in the search results. That would be entirely transparent to the user and would not depend on running any commands.

@pklinef what do you think?

Reply:
If workspace/query metacards are not migrated then all of the functionality for existing workspaces will not work.